src/main/java/org/openstreetmap/atlas/generator/tools/spark/persistence/PersistenceTools.java

```diff
@@ -43,8 +46,24 @@ public PersistenceTools(final Map<String, String> configurationMap)
 
     public CountryBoundaryMap boundaries(final String input)
     {
-        return CountryBoundaryMap.fromPlainText(SparkJob
-                .resource(SparkFileHelper.combine(input, BOUNDARIES_FILE), this.configurationMap));
+        try (ResourceCloseable resource = SparkJob
+                .resource(SparkFileHelper.combine(input, BOUNDARIES_FILE), this.configurationMap))
+        {
+            return CountryBoundaryMap.fromPlainText(resource);
+        }
+        catch (final Exception e)
+        {
+            logger.error(e);
+            if (e instanceof RuntimeException)
```

Comment:
I would remove this check, since CoreException is also a runtime exception. I would just wrap everything in the core exception below

Reply:
I wanted to avoid wrapping a RuntimeException in another exception (i.e., I wanted to avoid cases where there is a CoreException with 14 inner CoreExceptions).

Reply:
Just to clarify, did you still want me to change this, or can I leave this as is?